blog.md

```diff
@@ -1,26 +1,78 @@
+While an IaaS is not strictly necessary here (I could run Deis straight on bare metal) it makes sense to use something like [OpenStack](http://www.openstack.org/) as provides excellent APIs to spin up VMs.  I installed OpenStack across across a set of physical machines using [Blue Box's](https://www.blueboxcloud.com/) [Ursula](https://github.com/blueboxgroup/ursula).
+
+Next the PaaS itself, I have familiarity with [Deis](http://Deis.io/) already and I really like its (Heroku-esque) user experience.  I span up a three node Deis cluster on OpenStack using the terraform instructions [here](https://github.com/paulczar/Deis/tree/openstack_provision_script/contrib/openstack).
+
+I also span an additional three [CoreOS](https://coreos.com) nodes using [Terraform](https://terraform.io/) on which I ran [Jenkins](http://jenkins-ci.org/) using the standard [jenkins docker image](https://hub.docker.com/_/jenkins/).
+
+To demonstrate a backing service for my application I also ran a Percona database cluster (one on each of the CoreOS nodes) using the [paulczar/percona-galera](https://github.com/paulczar/docker-percona_galera) docker image which uses [etcd](https://coreos.com/etcd/) to find each other and form automatically create the galera replication set. I also deployed a mysql proxy using the [paulczar/maxscale](https://hub.docker.com/r/paulczar/maxscale/) image which discovers the databases via [etcd](https://coreos.com/etcd/).
+
+## Ghost
+
+The application I chose to demo is the [Ghost blogging platform](https://ghost.org/download/) (literally the blog you are reading right now). I chose it because it's a fairly simple app with well-known backing service (MySQL). The source including my `Dockerfile` and customizations can be found in the [paulczar/ci-demo](https://github.com/paulczar/ci-demo) github repository.
+
+The database backend for ghost is hosted on the percona cluster I have running on the CoreOS nodes and the location and credentials for it are passed into Ghost via environment variables set by the Deis cli when creating the applications.
+
+I wanted to stick to the [GitHub Flow](https://guides.github.com/introduction/flow/) for development as much as possible (the merge/deploy steps are a bit different here, but should be easy enough to follow along with) utilizing GitHub's notification system on Pull Requests and Merges to trigger Jenkins jobs.
+
+I created a `development` branch in the git repository for the application and then used the Deis cli to create two applications [ghost](http://ghost.ci-demo.paulcz.net) from the code in the `master` branch and [stage-ghost](http://stage-ghost.ci-demo.paulcz.net) from the development branch to be my Production and Staging environments.
+
+Both the `development` and `master` branches are protected with GitHub settings that restrict changes from being pushed directly to the branch and any Pull Requests need to pass tests before they can be merged.
```

Comment:
Kurtis Blow remains in the house:
« [...] directly to the branch**. Furthermore, any Pull Requests** [...] »